firedrake/preconditioners/patch.py

```diff
@@ -658,6 +691,7 @@ def initialize(self, obj):
 
         mesh = J.ufl_domain()
         self.plex = mesh._plex
+        self.plex.setAttr("__firedrake_mesh__", weakref.proxy(mesh))
```

Comment:
Add comment. Maybe add a `destroy` method to the PC class that removes this attribute.

Reply:
Petsc will call that destroy when the PC goes out of scope, so you can do teardown then.

Reply:
Great point, done.